xdsl/ir.py

```diff
@@ -503,8 +503,9 @@ def operands(self, new: list[SSAValue] | tuple[SSAValue, ...]):
         self._operands = new
 
     def __post_init__(self):
-        assert (self.name != "")
-        assert (isinstance(self.name, str))
+    if self.parent is not None:
```

Comment:
Seems that the indentation is off. Have you run the tests with `pytest` and `lit tests/filecheck` to make sure it all still works?